iquip/apps/monitor.py

```diff
@@ -108,27 +143,88 @@ def __init__(
             row, column = idx // numColumns, idx % numColumns
             self.ttlWidgets[name] = ttlWidget
             ttlWidgetLayout.addWidget(ttlWidget, row, column)
-        self.button = QPushButton("Not Overriding?", self)
-        self.button.setCheckable(True)
+        self.levelOnButton = QPushButton("ON", self)
+        # self.levelOnButton.setAlignment(Qt.AlignLeft)
+        self.levelOffButton = QPushButton("OFF", self)
+        # self.levelOffButton.setAlignment(Qt.AlignRight)
+        self.overrideButton = QPushButton("", self)
+        self.overrideButton.setCheckable(True)
         # layout
+        levelLayout = QHBoxLayout()
+        levelLayout.addWidget(self.levelOnButton)
+        levelLayout.addWidget(self.levelOffButton)
         layout = QVBoxLayout(self)
         layout.addLayout(ttlWidgetLayout)
-        layout.addWidget(self.button)
+        layout.addLayout(levelLayout)
+        layout.addWidget(self.overrideButton)
         # signal connection
-        self.button.clicked.connect(self.overrideChanged)
+        self.levelOnButton.clicked.connect(lambda: self.levelChangedRequested.emit(True))
+        self.levelOffButton.clicked.connect(lambda: self.levelChangedRequested.emit(False))
         self.overrideChanged.connect(self._setButtonText)
+        self.overrideButton.clicked.connect(
+            functools.partial(self.overrideButton.setEnabled, False)
+        )
+        self.overrideButton.clicked.connect(self.overrideChangedRequested)
 
     @pyqtSlot(bool)
     def _setButtonText(self, override: bool):
         """Sets the button text.
         
         Args:
-            override: Whether the button is now checked or not.
+            override: Whether the override value is on or off.
         """
         if override:
-            self.button.setText("Overriding")
+            self.overrideButton.setText("Overriding")
         else:
-            self.button.setText("Not Overriding")
+            self.overrideButton.setText("Not Overriding")
+        self.overrideButton.setChecked(override)
+        self.overrideButton.setEnabled(True)
+
+
+class _TTLStatusThread(QThread):
+    """QThread for fetching the TTL status from the proxy server.
+    
+    Signals:
+        fetched(status): The TTL status is fetched.
+          The "status" is a dictionary with three keys.
+            "outputs": A dictionary with a TTL name and its output value.
+            "levels": A dictionary with a TTL name and its level.
+            "overriding": The override value. If None, there is no change.
+    
+    Attributes:
+        url: The web socket url.
+        devices: The tuple of TTL names.
+    """
+
+    fetched = pyqtSignal(dict)
+
+    def __init__(self, ip: str, port: int, devices: Tuple[str], parent: Optional[QObject] = None):
```

Comment:
Did you intend `devices: Tuple[str, ...]`?

Reply:
Oh, you're right! However, I updated the type of devices to `list`, so I just use `List[str]`.